x-pack/solutions/security/plugins/security_solution/public/management/components/policy_selector/policy_selector.tsx

```diff
@@ -0,0 +1,748 @@
+   */
+  useCheckbox?: boolean;
+  /**
+   * If `true`, each policy option on the list will have a `view details` link. The link will point
+   * to the Security Solution policy details page if the policy is for elastic defend, and to the
+   * fleet package policy details if not.
+   */
+  showPolicyLink?: boolean;
+  /**
+   * Callback for setting additional display properties for each policy displayed on the list. callback is provided
+   * with the package policy from fleet
+   * @param policy
+   */
+  policyDisplayOptions?: (
+    policy: PackagePolicy
+  ) => Pick<EuiSelectableOption, 'disabled' | 'toolTipContent' | 'toolTipProps'>;
+  /** If `true`, then only a single selection will be allowed. Default is `false` */
+  singleSelection?: boolean;
+  /** Disable selector */
+  isDisabled?: boolean;
+  'data-test-subj'?: string;
+}
+
+/**
+ * Provides a component that displays a list of policies fetched from Fleet, which the user can
+ * select and unselect. By default, it queries Fleet for Elastic Defend policies, but that can be
+ * configured via `queryOptions.kuery`, thus it can display any type of Fleet integration policies
+ */
+export const PolicySelector = memo<PolicySelectorProps>(
+  ({
+    queryOptions: {
+      kuery = `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name: endpoint`,
+      sortField = 'name',
+      sortOrder = 'asc',
+      withAgentCount = false,
+      perPage = 20,
+    } = {},
+    selectedPolicyIds,
+    searchFields = ['name', 'description', 'policy_ids', 'package.name'],
+    onChange,
+    onFetch,
+    height,
+    useCheckbox = false,
+    showPolicyLink = false,
+    policyDisplayOptions,
+    singleSelection = false,
+    additionalListItems = [],
+    isDisabled = false,
+    'data-test-subj': dataTestSubj,
+  }) => {
+    const toasts = useToasts();
+    const getTestId = useTestIdGenerator(dataTestSubj);
+    const { getAppUrl } = useAppUrl();
+    const { canReadPolicyManagement, canWriteIntegrationPolicies } =
+      useUserPrivileges().endpointPrivileges;
+    const [page, setPage] = useState(1);
+    const [selectedListPage, setSelectedListPage] = useState(1);
+    const [userSearchValue, setUserSearchValue] = useState('');
+    const [searchKuery, setSearchKuery] = useState(kuery);
+    const [view, setView] = useState<'full-list' | 'selected-list'>('full-list');
+    const reactWindowFixedSizeList = useRef<{ scrollToItem: (index: number) => void }>();
+    const handledApiData = useRef(new WeakSet<GetPackagePoliciesResponse>());
+
+    // Delays setting the `searchKuery` value thus allowing the user to pause typing - so
+    // that we don't call the API on every character change.
+    useDebounce(
+      () => {
+        setPage(1);
+
+        if (userSearchValue) {
+          const kueryWithSearchValue: string = searchFields
+            .map((field) => `(${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.${field}:*${userSearchValue}*)`)
```

Comment:
maybe it would be worth to limit the search to only a small number of fields, e.g. policy name, description and id. now it looks a bit confusing when i try to search:
<img width="863" alt="image" src="https://github.com/user-attachments/assets/746b79af-5b71-4cf7-9c2e-2d2efa71a721" />